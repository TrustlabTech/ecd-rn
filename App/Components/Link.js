/**
 * Early Childhood Development App
 * @copyright 2016 Global Consent Ltd
 * Civvals, 50 Seymour Street, London, England, W1H 7JG
 * @author Werner Roets <werner@io.co.za>
 */

import React, { Component } from 'react'
import {
  View,
  Text,
  TouchableHighlight
} from 'react-native'

import { Colours, FontSizes } from '../GlobalStyles'
export default class Link extends Component {

  constructor(props) {
    super(props)
  }

  render() {
    return (
        <TouchableHighlight style={{padding: 5, margin: 2}} onPress={ this.props.onPress }>
          <Text style={{
            color: Colours.secondary,
            fontSize: FontSizes.p
          }}
        >
          {this.props.text}</Text>
        </TouchableHighlight>
    )
  }
}