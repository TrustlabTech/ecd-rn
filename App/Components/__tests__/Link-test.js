import 'react-native'
import React from 'react'
import Link from '../Link'

import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <Link text="Test link" onPress={() => ({}) }/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})