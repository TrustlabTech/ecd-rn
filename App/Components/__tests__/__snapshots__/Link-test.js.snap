exports[`test renders correctly 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  onPress={[Function]}
  style={
    Object {
      "margin": 2,
      "padding": 5,
    }
  }
  underlayColor="black">
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#5b5b5b",
        "fontSize": 16,
      }
    }>
    Test link
  </Text>
</TouchableHighlight>
`;
